### PR TITLE
Modified ThrottleBot to stress containers individually

### PR DESCRIPTION
**Pull request by TsaiAnson**
Previously, Throttlebot stresses all containers in a specific VM through cpu, disk, and network throttling. Now, Throttlebot stresses individual containers in a VM through cpu and disk throttling (Per-container network throttling has yet to be implemented). The changes in these commits include the implementation of a container_id argument for most stressing functions, the addition of an argument for cpu period in cpu stressing, as well as other minor fixes (syntax, indentations, duplicate functions).

### CODE REVIEW

File: src/modify_resources.py [hunk 1]
```diff
@@ -59,62 +55,66 @@ def set_network_bandwidth(ssh_client, container_to_bandwidth, outgoing_traffic=T
     for container in container_to_bandwidth:
         bandwidth = int(container_to_bandwidth[container])
 
-                #Temporary!
+                # Temporary!
         container = 'ens3'
         cmd1 = 'sudo tc qdisc add dev {} handle 1: root htb default 11'.format(container)
         cmd2 = 'sudo tc class add dev {} parent 1: classid 1:1 htb rate {}kbit ceil {}kbit'.format(container, bandwidth, bandwidth)
         cmd3 = 'sudo tc class add dev {} parent 1:1 classid 1:11 htb rate {}kbit ceil {}kbit'.format(container, bandwidth, bandwidth)
-                #UNNECESSARY BECAUSE MICHAEL IS A FUCKING DUMBASS!!!!
```
Comment (mchang6137) [inline src/modify_resources.py:67]: Please remove this line. Thanks :)

File: src/stress_scheduler.py [hunk 2]
```diff
@@ -27,66 +27,66 @@
 
 ### Start the stresses on the various resources
 
-def start_causal_cpu(ssh_client, disk_rate, network_rate):
-    throttle_network(ssh_client, network_rate)
-    return throttle_disk(ssh_client, disk_rate)
+def start_causal_cpu(ssh_client, container_id, disk_rate, network_rate):
```
Comment (mchang6137) [inline src/stress_scheduler.py]: For all the functions here, keep in mind that that just passing in the argument container_id to this helper stress functions probably makes that function redundant. I'll accept it for now but your next PR will probably need to change this.

File: src/stress_scheduler.py [hunk 3]
```diff
@@ -159,9 +161,11 @@ def explore_stress_space(ssh_client, resource, experiment_args, experiment_type,
         print 'Invalid resource'
         return
 
+# Note: Shelved for later
```
Comment (mchang6137) [inline src/stress_scheduler.py]: Just remove this function.

**Review by mchang6137 — COMMENTED** [review]
Most of this looks good! I left a couple of comments but you the high level point is that you may need to change some of these interfaces between functions.

File: src/modify_resources.py (hunk 1 above)
Comment (TsaiAnson) [reply src/modify_resources.py:67]: Yup, it has been removed below!

**Comment by hantaowang** [comment]
I see that you have it set up right now so that the main method in stress_scheduler takes in a specific container id. This is probably fine for now, but I think either in this pull request or a future one you should change it so that stress_scheduler takes in an optional file (i.e containers.conf) that is a user defined whitelist and blacklist.

The conf file would be parsed into some kind of object with a method to validate containers. For example, the object could be called `container_list` with a method `container_list.validate(cont_ID)`.  In addition, the object also has a default blacklist of all containers that have the names etcd, cadvisor, ovos, etc. This object would be passed into either model_machine or one of the specific stressing methods so that when you iterate through all the containers you only run the test if validate returns true for that container. 

This object may be useful for multiple iterations of testing as stress scheduler could output another `container_list` object based on the result of the stress that would be used again to narrow down the search.

**Comment by mchang6137** [comment]
LGTM

**Pull request merged**